### PR TITLE
 docs(specs): components that requires positioning

### PR DESCRIPTION
**Pull request by cooper-joe**
First pass at specs for each component that requires positioning. A generic `position-options.feature` specs the available options/locations for positioning. This follows the [position utility discussion](https://github.com/dhis2/notes/blob/master/decisions/2019/10/15-popover-position.md).

@dhis2/team-apps, please add to and correct where necessary! I tried to keep the specs generic enough to apply to all situations, but with enough detail that they can actually be tested.

---

### Notes

I have covered the components that definitely need the position utility. There are also some 'maybe' components. These can 'reuse' the positioning in the specs, so do not have spec files.

- DropdownButton: use `menu` 
- SplitButton: use `menu`
- HeaderBar App menu: use `menu`
- HeaderBar Profile menu: use `menu`

### CODE REVIEW

File: cypress/integration/position/menu.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+And the menu is right aligned with the anchor
+
+Scenario: Forced body overflow
+Given the anchor has less than 128px space to the left and the right, before the limit of the <body> element
+When the menu is rendered
+Then the menu is rendered below the anchor
+And the menu is left aligned with the anchor
+And the menu overflows the body element
+// ¯\_(ツ)_/¯
+// The app would need to handle this, this is beyond the scope of the positioning utility
```
Comment (cooper-joe) [inline cypress/integration/position/menu.feature]: I think this is beyond the scope of the position utility. Apps should have a way to deal with overflowing elements.

**Comment by Mohammer5** [comment]
The folder structure is flipped, it should be:

```
cypress/integration/Menu/position.feature
cypress/integration/Popover/position.feature
cypress/integration/SelectMenu/position.feature
...
````

Have a look how I named the feature files of the AlertBar. They have the same name as the feature string itself. That makes it easier to find the right file for a certain feature. It's also what PEPFAR is doing

**Comment by Mohammer5** [comment]
@cooper-joe I've corrected the indentations and added the `Background` keyword.
I'd like to change some sentences, but I won't do that on this branch. I'll prepare a PR for your branch so we can talk about it (and most likely correct my English mistakes)

**Comment by Mohammer5** [comment]
@cooper-joe I've created a `Positioning` branch and [PR](https://github.com/dhis2/ui-core/pull/655) so we don't put these files on master

**Comment by cooper-joe** [comment]
I've rebased this branch and squashed the commits, @Mohammer5 could you rereview before I merge?

**Comment by Mohammer5** [comment]
@cooper-joe Looks good to me!

**Pull request merged**